src/Data/Argonaut/Decode/Generic/Rep.purs

```diff
@@ -87,3 +91,32 @@ genericDecodeJson = map Rep.to <<< decodeRep
 
 mFail :: forall a. String -> Maybe a -> Either String a
 mFail msg = maybe (Left msg) Right
+
+-- | A function for decoding `Generic` sum types using string literal representations
+-- | Takes a function for transforming the tag name in encoding
+decodeLiteralSum :: forall a r. Rep.Generic a r => DecodeLiteral r => (String -> String) -> Json -> Either String a
+decodeLiteralSum tagNameTransform = map Rep.to <<< decodeLiteral tagNameTransform
```

Comment:
I guess that's what I'm not understanding perhaps... why would you need to do that? If you're using generics it seems like it's something you would have serialized in the first place, so the specific encoding should be the same anyway?

It just seems a bit odd to me, as if you're using this argument you'd probably need an iso anyway, so you could just combine the encode/decode with a `map` as appropriate.